nemo/collections/nlp/data/language_modeling/megatron/megatron_batch_samplers.py

```diff
@@ -226,7 +216,7 @@ def __len__(self) -> int:
             When `rampup_batch_size` is enabled, the return value can be not exactly precise.
 
         """
-        active_total_samples = self.total_samples - self.last_batch_size
+        active_total_samples = self.total_samples - (self.last_batch_size if self.drop_last else 0)
```

Comment:
Same comment as above on the fact that the epoch in `__iter__()` won't be correct when `drop_last` is False and a comment may be needed.

Reply:
I've added the new raise exception so you never end up running with `drop_last=False` and MBS * DP > 1